code/game/objects/items/melee/baton.dm

```diff
@@ -519,9 +527,19 @@
 
 /obj/item/melee/baton/security/attack_self(mob/user)
 	if(cell?.charge >= cell_hit_cost)
-		active = !active
-		balloon_alert(user, "turned [active ? "on" : "off"]")
-		playsound(src, SFX_SPARKS, 75, TRUE, -1)
+		if(SSsecurity_level.get_current_level_as_number() >= SEC_LEVEL_RED)
+			active = !active
+			balloon_alert(user, "turned [active ? "on" : "off"]")
+			playsound(src, SFX_SPARKS, 75, TRUE, -1)
+		else
+			//Allow turning it off
+			if (!active)
+				to_chat(user, span_warning("You cannot use your baton in a non emergency situation"))
+			else
+				active = FALSE
+				balloon_alert(user, "turned [active ? "on" : "off"]")
+				playsound(src, SFX_SPARKS, 75, TRUE, -1)
```

Comment:
This proc is desperately crying out for early returns
We should have a setter for active that checks if the cell has charge/exists, and only plays the sound if it does

Would make this significantly less stupid